### PR TITLE
Refactor Trame widget - Vue 2 and Vue 3 versions

### LINKED ISSUE #4776: Option to use Vuetify 3 in Trame widgets
### Describe the feature you would like to be added.

Trame is migrating its default client type from vue 2 to vue 3 in December 2023 (see item 2 in [this discussion](https://github.com/Kitware/trame/discussions/295)). The current implementation of the trame widgets in `pyvista/trame/ui.py` uses vuetify 2, which is incompatible with the vue 3 client type.

### Pseudocode or Screenshots

Switching over can be as simple as changing the import. In my development version, adding the following lines to `pyvista/trame/ui.py` makes my simple app compatible with the vue 3 client type:
```python
from trame.app import get_server

CLIENT_TYPE = get_server().client_type
if CLIENT_TYPE == 'vue2':
    from trame.widgets import vuetify
else:
    from trame.widgets import vuetify3 as vuetify
```

However, a more robust approach may be desired by the pyvista team. The above "quick fix" has some limitations:
- The trame server client type must be set before the desired widgets are imported
- The vuetify API has slight differences between vuetify 2 and 3. Here are a couple I noticed within the file:
    - `dense=True` -> `density='compact'`
    - `on_icon` -> `true_icon`
    - `off_icon` -> `false_icon`
- The default spacing with vuetify is also slightly different (see below screenshot of plotter controls)
image

### PR DESCRIPTION
**Pull request by annehaley**
### Overview

This PR is intended to make the Trame widget compatible with both Vue 2 and Vue 3 client types within a Trame application. Resolves #4776. 
This PR refactors `pyvista/trame/ui.py` into several new files: 

- `trame/pyvista/ui/__init__.py`
- `trame/pyvista/ui/base_viewer.py`
- `trame/pyvista/ui/vuetify2.py`
- `trame/pyvista/ui/vuetify3.py`

### Details
Previously, `ui.py` defined a class `Viewer` and several top-level functions and variables. The following changes were made:

- Most top-level functions and variables were moved to `ui/__init__.py`. 
- Two top-level functions, `button` and `checkbox` required different implementations for the Vuetify 2 and Vuetify 3 API, so these were moved to `ui/vuetify2.py` and `ui/vuetify3.py`, respective to the implementations.
- The `Viewer` class was renamed to `BaseViewer` and moved to `ui/base_viewer.py`. Two class methods, `ui_controls` and `ui`, required different implementations for Vuetify 2 and Vuetify 3. The `ui` method for `BaseViewer` now raises a `NotImplementedError` and the `ui_controls` method was only ever referenced within the `ui` method, so it was removed entirely from `BaseViewer`
- Two new `Viewer` classes were written in `ui/vuetify2.py` and `ui/vuetify3.py`, both derived from the `BaseViewer` class. Both derived classes implement the `ui` and `ui_controls` methods, using the Vuetify 2 and Vuetify 3 APIs respectively.
- The top-level function `get_or_create_viewer` (used within the top-level function `plotter_ui`) now has an extra bit of logic to determine which implementation of `Viewer` to use in the case of new viewer creation. See below code. 

```
CLIENT_TYPE = get_server().client_type
if CLIENT_TYPE == 'vue2':
    return Vue2Viewer(plotter, suppress_rendering=suppress_rendering)
else:
    return Vue3Viewer(plotter, suppress_rendering=suppress_rendering)
```

This solves the problem I mentioned in the linked issue about needing to set the Trame server's client type before the import of `plotter_ui` within the Trame application. Now the client type is not fetched until the `plotter_ui` function calls `get_or_create_viewer`. This PR does not change the API for using the `plotter_ui` function.

### CODE REVIEW

File: pyvista/trame/ui/vuetify2.py
Comment (akaszynski) [inline pyvista/trame/ui/vuetify2.py:55]: These should still all be indented to get it to render right in sphinx.

Comment (annehaley) [reply pyvista/trame/ui/vuetify2.py]: Thank you. Fixed in [8bcd586](https://github.com/pyvista/pyvista/pull/4811/commits/8bcd586cdfdd9289eb73b2ef22e1032bf228405b). Let me know if there are any I missed.

File: pyvista/trame/ui/__init__.py
Comment (tkoyama010) [inline pyvista/trame/ui/__init__.py:89]:
```suggestion
    PyVistaRemoteLocalView | PyVistaRemoteView | PyVistaLocalView
```

File: pyvista/trame/ui/vuetify2.py
Comment (tkoyama010) [inline pyvista/trame/ui/vuetify2.py:192]:
```suggestion
            using the ``'trame'`` mode.
```

File: pyvista/trame/ui/__init__.py
Comment (tkoyama010) [inline pyvista/trame/ui/__init__.py:100]:
```suggestion
    viewer = get_viewer(
```

File: pyvista/trame/ui/__init__.py
Comment (tkoyama010) [inline pyvista/trame/ui/__init__.py:15]:
```suggestion
from .vuetify3 import Viewer as Vue3Viewer
import warnings
```

File: pyvista/trame/ui/__init__.py
Comment (tkoyama010) [inline pyvista/trame/ui/__init__.py:48]:
```suggestion
            warnings.warn("Suppress rendering on the plotter is changed to " + str(suppress_rendering), UserWarning)
```

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Review by tkoyama010 — COMMENTED** [review]
In my mind, the meaning of `create` in `get_or_create_viewer` is encompassed by `get`. Therefore, I have modified the name to `get_viewer`.

**Review by tkoyama010 — APPROVED** [review]
LGTM. We are glad that can use Vue3 in PyVista viewer.

**Comment by annehaley** [comment]
Thank you @tkoyama010. I will leave it to your discretion when to merge, as I do not have authorization to complete the merge.

**Comment by tkoyama010** [comment]
We have a rule of waiting 24 hours after approve to give maintainers around the world the opportunity to comment. Please wait another 10 hours.

**Comment by tkoyama010** [comment]
I would like to add #4639 to the release if possible. Could you two please help me?

**Pull request merged**
